docs/proposals/proposal-trigger-and-deploy.md

```diff
@@ -0,0 +1,247 @@
+# GRP-001 - Trigger and Deploy GitHub Action workflow from an upstream CNCF project
```

Comment:
@dipankardas011 Little nitpick but please rename the file to start `proposal-001-`